### PR TITLE
create models with custom methods

### PR DESCRIPTION
**Pull request by petushka1**
# Milestone - 2
## Create Models and custom methods

- [x] Rollback Migrations and fix typo
- [x] Run migrations and generate tables
- [x] Write Models
- [x] Insert data in tables
- [x] Modify Models by adding custom methods
- [x] Check all custom methods through `bin/rails console`        
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  def three_most_recent_records
+    posts.order(created_at: :desc).first(3)
+  end
+end
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] a user should be able to have many comments and likes as well similar to the posts.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  after_create :increment_posts_counter
+
+  def increment_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def five_recent_comments
+    comments.order(created_at: :desc).first(5)
+  end
+end
```
Comment (mikethreels) [inline app/models/post.rb]: - [ ] a post should be able to have many likes as well, similar to the comments.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :author, class_name: 'User'
+end
```
Comment (mikethreels) [inline app/models/comment.rb]: - [ ] The Comment model should have a method that updates the comments counter for a post.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Nadia, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by petushka1** [comment]
- add associations
- add increment_comments_counter method
- check counter update
image

**Review by Davidosky007 — APPROVED** [review]
Hi @petushka1 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All associations created properly 👍 
* No linter errors 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**